core/src/main/scala/org/apache/spark/deploy/SparkCuratorUtil.scala

```diff
@@ -35,8 +35,11 @@ private[spark] object SparkCuratorUtil extends Logging {
   def newClient(
       conf: SparkConf,
       zkUrlConf: String = "spark.deploy.zookeeper.url"): CuratorFramework = {
```

Comment:
It's an unnecessary abstraction that makes the code more complicated. We can always introduce it later if we actually need it.